### PR TITLE
add aksarc module

### PR DESCRIPTION
**Pull request by ArulvelMuralidharan**
## Description

Add AKSArc powershell module

##ChangeLog Description
 Adds the AKSArc powershell module which allows users to create/manage on premise kubernetes clusters via Azure Cloud. 

## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/AksArc/AksArc.Autorest/custom/Get-AzAksArcCluster.ps1 [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+Gets the provisioned cluster instance
+.Example
+{{ Add code here }}
+.Example
+{{ Add code here }}
+
+.Outputs
+Microsoft.Azure.PowerShell.Cmdlets.AksArc.Models.IProvisionedCluster
+.Link
+https://learn.microsoft.com/powershell/module/az.aksarc/get-azaksarcprovisionedclusterinstance
```
Comment (NoriZC) [inline src/AksArc/AksArc.Autorest/custom/Get-AzAksArcCluster.ps1:30]: The online versions in help markdown come from the `Link` session here in custom files. Please correct the links as well as the Synopsis/ Description/ Example/ Outputs for each cmdlet.

**Review by NoriZC — CHANGES_REQUESTED** [review]
Please fix the Static Analysis Help markdown issue and fix the failed tests.

File: src/AksArc/AksArc.Autorest/custom/Get-AzAksArcCluster.ps1 [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+param(
+    [Parameter(Mandatory)]
+    [Alias('Name')]
+    [Microsoft.Azure.PowerShell.Cmdlets.AksArc.Category('Path')]
+    [System.String]
+    # The name of the Kubernetes cluster on which get is called.
+    ${ClusterName},
+
+    [Parameter(Mandatory)]
+    [Alias('resource-group')]
```
Comment (NoriZC) [inline src/AksArc/AksArc.Autorest/custom/Get-AzAksArcCluster.ps1:42]: Why would you have the Alias which do not match Azure PowerShell naming conventions?

File: src/AksArc/AksArc.Autorest/custom/Get-AzAksArcCluster.ps1 [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+function Get-AzAksArcCluster {
+[OutputType([Microsoft.Azure.PowerShell.Cmdlets.AksArc.Models.IProvisionedCluster])]
+[CmdletBinding(DefaultParameterSetName='Get', PositionalBinding=$false)]
+param(
+    [Parameter(Mandatory)]
+    [Alias('Name')]
+    [Microsoft.Azure.PowerShell.Cmdlets.AksArc.Category('Path')]
+    [System.String]
+    # The name of the Kubernetes cluster on which get is called.
+    ${ClusterName},
```
Comment (NoriZC) [inline src/AksArc/AksArc.Autorest/custom/Get-AzAksArcCluster.ps1:39]:
```suggestion
    [Parameter(Mandatory)]
    [Alias('Name')]
    [Microsoft.Azure.PowerShell.Cmdlets.AksArc.Category('Path')]
    [System.String]
    # The name of the Kubernetes cluster on which get is called.
    ${ClusterName},
```
```suggestion
    [Parameter(Mandatory)]
    [Alias('Name')]
    [Microsoft.Azure.PowerShell.Cmdlets.AksArc.Category('Path')]
    [System.String]
    # The name of the Kubernetes cluster on which get is called.
    ${ClusterName},
```
```suggestion
    [Parameter(Mandatory)]
    [Microsoft.Azure.PowerShell.Cmdlets.AksArc.Category('Path')]
    [System.String]
    # The name of the Kubernetes cluster on which get is called.
    ${Name},
```

File: src/AksArc/AksArc.Autorest/custom/Get-AzAksArcCluster.ps1 [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+    [Parameter()]
+    [Microsoft.Azure.PowerShell.Cmdlets.AksArc.Category('Path')]
+    [Microsoft.Azure.PowerShell.Cmdlets.AksArc.Runtime.DefaultInfo(Script='(Get-AzContext).Subscription.Id')]
+    [System.String]
+    # The ID of the target subscription.
+    ${SubscriptionId},
+
+    [Parameter()]
+    [Alias('AzureRMContext', 'AzureCredential')]
```
Comment (NoriZC) [inline src/AksArc/AksArc.Autorest/custom/Get-AzAksArcCluster.ps1:57]: Same here, alias is not necessary to avoid maintanance effort.

**Comment by NoriZC** [comment]
Hi @ArulvelMuralidharan notified that you set a lot of alias for parameters, could you explain where those alias come from and why are they necessary. Although Azure PowerShell commands support Alias, usually we won't suggest to have many of them. Because the maintanance cost would be a burden. Meanwhile too many names for new parameters will also cause confusion.

File: src/AksArc/AksArc.Autorest/custom/Get-AzAksArcCluster.ps1 (hunk 3 above)
Comment (ArulvelMuralidharan) [reply src/AksArc/AksArc.Autorest/custom/Get-AzAksArcCluster.ps1]: Thanks resolved

File: src/AksArc/AksArc.Autorest/custom/Get-AzAksArcCluster.ps1 (hunk 2 above)
Comment (ArulvelMuralidharan) [reply src/AksArc/AksArc.Autorest/custom/Get-AzAksArcCluster.ps1]: Thanks resolved. Just wanted to make it more similar to our cli extension but it isn't necessary.

**Comment by ArulvelMuralidharan** [comment]
@NoriZC Thanks for reviewing the pr! I updated the pr with the changes. It is now passing the test cases as well. Are we good to merge this pr?

**Review by NoriZC — APPROVED** [review]

**Pull request merged**